tests/providers/aws/aws_cli.py

```diff
@@ -0,0 +1,167 @@
+    logger.info('CLI command: {0:s}'.format(cmd))
+    try:
+      output = subprocess.check_output(
+          cmd.split(), stderr=subprocess.STDOUT, shell=False)
+      logger.info(output)
+      try:
+        return self.aws.ec2.GetInstancesByName(vm_name)[0]
+      except ResourceNotFoundError as error:
+        raise ResourceNotFoundError(
+            'Failed finding created VM in account profile {0:s}'.format(
+                self.aws.aws_profile), __name__) from error
+    except subprocess.CalledProcessError as error:
+      raise ResourceCreationError(
+          'Failed creating VM in account profile {0:s}'.format(
+              self.aws.aws_profile), __name__) from error
+
+  def CreateVolumeCopy(
+      self,
+      zone: str,
+      dst_zone: Optional[str] = None,
+      instance_id: Optional[str] = None,
+      volume_id: Optional[str] = None) -> ebs.AWSVolume:
+    """Create a volume copy.
+    Args:
+      zone (str): The AWS zone in which the volume is located, e.g.
+          'us-east-2b'.
+      dst_zone (str): Optional. The AWS zone in which to create the volume
+        copy. By default, this is the same as 'zone'.
+      instance_id (str): Optional. Instance ID of the instance using the volume
+          to be copied. If specified, the boot volume of the instance will be
+          copied. If volume_id is also specified, then the volume pointed by
+          that volume_id will be copied.
+      volume_id (str): Optional. ID of the volume to copy. If not set,
+          then instance_id needs to be set and the boot volume will be copied.
+
+    Returns:
+      ebs.AWSVolume: A ebs.AWSVolume object that represents the volume
+          copy.
+
+    Raises:
+      ResourceCreationError: If the volume could not be created.
+      ResourceNotFoundError: If the created volume copy could not be found.
+    """
+    cmd = 'cloudforensics aws {0:s} copydisk'.format(zone)
+    if instance_id:
+      cmd += ' --instance_id={0:s}'.format(instance_id)
+    elif volume_id:
+      cmd += ' --volume_id={0:s}'.format(volume_id)
+    if dst_zone:
+      cmd += ' --dst_zone={0:s}'.format(dst_zone)
+    logger.info('CLI command: {0:s}'.format(cmd))
+    try:
+      output = subprocess.check_output(
+          cmd.split(), stderr=subprocess.STDOUT, shell=False)
+      if not output:
+        raise ResourceCreationError(
+            'Could not find volume copy result', __name__)
+      volume_copy_name = output.decode('utf-8').split(' ')[-1]
+      volume_copy_name = volume_copy_name[:volume_copy_name.rindex('copy') + 4]
+      logger.info(output)
+      logger.info("Volume successfully copied to {0:s}".format(
+          volume_copy_name))
+      try:
+        return self.aws.ebs.GetVolumesByName(volume_copy_name)[0]
+      except ResourceNotFoundError as error:
+        raise ResourceNotFoundError(
+            'Failed finding copied volume in account profile {0:s}'.format(
+                self.aws.aws_profile), __name__) from error
```

Comment:
Same thing here, and in general. I would limit these helper functions (e.g. `CreateVolumeCopy`) to just creating the command you need, and then do all the checking as part of the test itself.

Reply:
ACK, will update